src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -41,9 +41,19 @@ export const monoPlatform: Platform = {
 
   findMethod: findMethod,
 
-  callEntryPoint: function callEntryPoint(assemblyName: string): System_Object {
-    const empty_array = mono_obj_array_new(0);
-    return mono_call_assembly_entry_point(assemblyName, [empty_array]);
+  callEntryPoint: function callEntryPoint(assemblyName: string): Promise<void> {
+    // Find entrypoint
+    const assembly = findAssembly(assemblyName);
+    const entrypointMethodHandle = assembly_get_entry_point(assembly);
+
+    // Since no .NET code has run yet, manually initialize JS interop
+    const ensureJsInteropInitializedMethod = this.findMethod('Microsoft.AspNetCore.Blazor', 'Microsoft.AspNetCore.Blazor.Services', 'WebAssemblyJSRuntime', 'EnsureInitialized');
+    this.callMethod(ensureJsInteropInitializedMethod, null, []);
```

Comment:
Not that we allow it today, because this will make it impossible for users to do anything that configures JS Interop in their main method (before "blazor startup"). Does it make sense for us to use unmarshalled interop for these kinds of things?

Reply:
I realize the reason you probably went this route is for the async stuff 😆

Reply:
Yes, I did do this with unmarshalled interop originally, but then switched to JSInterop when it seemed necessary to get all the async-Promise mapping stuff.

However, as per my comments on your startup APIs gist, I'd now like to change this design. If the main method becomes async and runs for the lifetime of the app, then it no longer makes sense for the `Promise` here to reflect the lifetime of the main method. 

Instead, `Blazor.start` should *only* represent the startup process and not the execution of the main method. This makes it useful for things like "stop showing a progress bar once the .NET code starts running". In the case where someone does want the JS side to response once the Blazor app completes entirely (e.g., due to an unhandled exception), that's unusual but can be achieved quite conveniently through JS interop.

So with the updated design it won't be necessary to use JS interop in the code here, and I'll go back to the simpler unmarshalled interop.